docs/Streams.md

```diff
@@ -74,38 +74,100 @@ If a stream gets canceled because it is in 'cancel on loss' mode, a `QUIC_STREAM
+
+### App-owned buffer mode
+
+App-owned buffer mode is a per-stream option allowing the application to provide its own receive memory buffers.
+Enabling app-owned mode is done differently depending on whether the stream is created locally or from the peer and is discussed below.
+
+When in app-owned mode, the application can call [`StreamProvideReceiveBuffers`](./????.md) to provide a list of memory buffers to MsQuic.
+[`StreamProvideReceiveBuffers`](./????.md) can be called at any time on a valid stream in app-owned mode, potentially inline from a notification handler.
+If called several times, the buffer provided through subsequent calls are added to the list.
+
+MsQuic will fill the provided buffer with received data, in the order they have been provided. Receive notification will be emitted as normal,
+indicating a list of `QUIC_BUFFER` pointing to the application provided buffers.
+Note that up to the number of buffers the application provided can be indicated at once, and that only part of a buffer can be indicated.
+There is no guarantees the `QUIC_BUFFER`s indicated in a receive notification will match the ones the application provided.
+
+The application is responsible for tracking the amount of data received and when a buffer it provided has been fully used.
+The application regains full ownership of a buffer after it get a receive notification for all bytes in the buffer and accept them by calling [`StreamMultiReceiveEnabled`](./Settings.md).
+If the application accepts all the buffer's bytes **inline** from the receive notification, by returning `QUIC_STATUS_SUCCESS` and setting `TotalBufferLength` appropriately,
+it can free or reuse the buffer while in the notification handler.
+
+For an application, providing receive buffers can improve performances by saving a copy: MsQuic places data directly in its final destination.
+However, it comes with a large complexity overhead for the application, both in term of memory management and in term of flow control:
+an application providing to much or too little buffer space could negatively impact performances.
+Because of this, app-owned mode should be considered an advanced feature and used with caution.
+
+**Note**: As of now, app-owned buffer mode is not compatible with multi-receive mode. If multi-receive mode is enabled for the connection and app-owned mode is enabled on a stream, that specific stream will behave as if multi-receive mode was disabled. This may change in the future.
+
+#### For a stream initiated by the application
+
+To use app-owned buffers on a locally created stream, the flag `QUIC_STREAM_OPEN_FLAG_APP_OWNED_BUFFERS` must be provided to the [`StreamOpen`](./api/StreamOpen.md).
+
+Before starting the stream with [`StreamStart`](./api/StreamStart.md), the application should call [`StreamProvideReceiveBuffers`](./api/StreamProvideReceiveBuffers.md) to provide some initial buffers.
```

Comment:
"should call"... this is not a _must call_?

Reply:
Nothing will prevent you from starting a stream in App-owned mode without providing buffers, and everything will be fine as long as you don't receive any data on it (or provide buffers before you do receive data). So technically, it's a "should", but it would be hard to find a reasonable scenario where it makes sense not to provide buffers before starting the connection.

I can make this a must if you think it makes it less confusing.